libraries/tracker-core/src/payload.ts

```diff
@@ -151,16 +168,61 @@ export function payloadBuilder(): PayloadBuilder {
  * @returns The request builder, with add and build methods
  */
 export function payloadJsonProcessor(encodeBase64: boolean): JsonProcessor {
-  return (payloadBuilder: PayloadBuilder, jsonForProcessing: EventJson) => {
-    for (const json of jsonForProcessing) {
-      const str = JSON.stringify(json.json);
+  return (
+    payloadBuilder: PayloadBuilder,
+    jsonForProcessing: EventJson,
+    contextEntitiesForProcessing: SelfDescribingJson[]
+  ) => {
+    const add = (json: any, keyIfEncoded: string, keyIfNotEncoded: string): void => {
+      const str = JSON.stringify(json);
       if (encodeBase64) {
-        payloadBuilder.add(json.keyIfEncoded, base64urlencode(str));
+        payloadBuilder.add(keyIfEncoded, base64urlencode(str));
+      } else {
+        payloadBuilder.add(keyIfNotEncoded, str);
+      }
+    };
+
+    let context = undefined;
+    const getContextFromPayload = () => {
+      let payload = payloadBuilder.getPayload();
+      if (encodeBase64 ? payload.cx : payload.co) {
+        return JSON.parse(encodeBase64 ? base64urldecode(payload.cx as string) : (payload.co as string));
+      }
+      return undefined;
+    };
+
+    for (const json of jsonForProcessing) {
+      if (json.keyIfEncoded == 'cx') {
+        context = getContextFromPayload();
```

Comment:
Just making a note for myself that this is a bug – overwriting the `context` variable in a for loop. Need to add a test for calling `addContextEntity` multiple times after each other.